data/backstage/infra/2025-10-30-how-the-opam-repository-works.md

```diff
@@ -0,0 +1,105 @@
+
+### The CI Pipeline
+
+At the heart of the repository's reliability is an extensive continuous integration system built around [OCurrent](https://ocurrent.org), which powers the [opam-repo-ci service](https://opam.ci.ocaml.org). When you submit a pull request to add or update a package, this system:
+
+- **Validates package metadata** using opam's built-in linting tools
+- **Tests compilation** across multiple OCaml compiler versions (currently supporting back to OCaml 4.08)
+- **Verifies installation** on various platforms including Linux distributions, macOS, FreeBSD, Windows, and multiple CPU architectures (x86_64, ARM64, ARM32, PowerPC, s390x, RISC-V)
+- **Checks reverse dependencies** to ensure your changes don't break existing packages
+
+This comprehensive testing matrix helps maintain the repository's high quality standards. Windows support was added in 2024, marking a significant expansion of the platform coverage.
+
+### Supporting Infrastructure
+
+The opam repository integrates with several other OCaml infrastructure services:
+
+- **Docker Base Images**: The [docker-base-images service](https://images.ci.ocaml.org) provides consistent testing environments, building official images for various Linux distributions, OCaml versions, and architectures
+- **Documentation CI**: Automatically builds and publishes package documentation to the [OCaml.org website's package area](https://ocaml.org/packages)
+- **Health Check Services**: Multiple services continuously monitor package installability across different platforms and compiler versions:
+  - [check.ci.ocaml.org](https://check.ci.ocaml.org): Tests all packages regularly
+  - [dune.check.ci.dev](https://dune.check.ci.dev): Tests with the Dune build system
+  - [windows.check.ci.dev](https://windows.check.ci.dev): Windows-specific testing
+  - [freebsd.check.ci.dev](https://freebsd.check.ci.dev): FreeBSD compatibility
+  - [oxcaml.check.ci.dev](https://oxcaml.check.ci.dev): Tests with OxCaml compiler variant
+
+## Governance and Curation Policies
+
+Unlike many package repositories that operate on a "publish first, moderate later" model, the opam repository follows a curated approach with explicit policies governing package inclusion and maintenance.
+
+### Core Principles
+
+The repository maintains quality through several key policies documented in the [repository wiki](https://github.com/ocaml/opam-repository/wiki):
+
+**Package Utility**: Packages must provide substantial utility to the OCaml community. Single-function modules or packages with minimal functionality are generally rejected to preserve namespace and repository efficiency.
+
+**Naming and Security**: Package names are carefully reviewed to prevent confusion with existing packages and potential security risks. Name squatting and packages with deceptively similar names to established libraries are not permitted.
```

Comment:
Perhaps we should just remove this whole point? We are concerned with security, but I am not sure we can claim this is a principle we are able to enforce. We just use best-effort, and people should not have false sense of security.